internal/host/validator.go

```diff
@@ -890,41 +890,50 @@ func (v *validator) sufficientOrUnknownInstallationDiskSpeed(c *validationContex
 	return ValidationFailure, "While preparing the previous installation the installation disk speed measurement failed or was found to be insufficient"
 }
 
-func (v *validator) hasSufficientNetworkLatencyRequirementForRole(c *validationContext) (ValidationStatus, string) {
-	if c.infraEnv != nil {
-		return ValidationSuccessSuppressOutput, ""
-	}
-	if len(c.cluster.Hosts) == 1 || c.clusterHostRequirements.Total.NetworkLatencyThresholdMs == nil || common.GetEffectiveRole(c.host) == models.HostRoleAutoAssign || hostutil.IsDay2Host(c.host) {
-		// Single Node use case || no requirements defined || role is auto assign
-		return ValidationSuccess, "Network latency requirement has been satisfied."
-	}
-	if len(c.host.Connectivity) == 0 {
-		return ValidationPending, "Missing network latency information."
-	}
-	s, hostLatencies, err := v.validateNetworkLatencyForRole(c.host, c.clusterHostRequirements, c.cluster.Hosts, c.inventoryCache)
-	if s == ValidationFailure {
-		if err != nil {
-			return ValidationFailure, fmt.Sprintf("Error while attempting to validate network latency: %s", err)
-		}
-		return ValidationFailure, fmt.Sprintf("Network latency requirements of %s %.2f ms not met for connectivity between %s and%s.", comparisonBuilder(*c.clusterHostRequirements.Total.NetworkLatencyThresholdMs), *c.clusterHostRequirements.Total.NetworkLatencyThresholdMs, c.host.ID, strings.Join(hostLatencies, ","))
-	}
-	if s == ValidationError {
-		return ValidationError, "Parse error while attempting to process the connectivity report"
+type HostTimingMetric struct {
+	HostName     string
+	TimingMetric float64
+}
+
+func (v *validator) summarizeHostTimingMetrics(packetLossInfo []HostTimingMetric) string {
+	result := []string{}
+	for _, p := range packetLossInfo {
+		result = append(result, fmt.Sprintf("%s (%.2f ms)", p.HostName, p.TimingMetric))
 	}
-	return ValidationSuccess, "Network latency requirement has been satisfied."
+	return strings.Join(result, ", ")
 }
 
-func (v *validator) validateNetworkLatencyForRole(host *models.Host, clusterRoleReqs *models.ClusterHostRequirements, hosts []*models.Host, inventoryCache InventoryCache) (ValidationStatus, []string, error) {
+type ThresholdTestType int
+
+const (
+	ThresholdTestL3AverageRTTMs ThresholdTestType = 0
+	ThresholdTestL3PacketLoss   ThresholdTestType = 1
+)
+
+func (v *validator) validatePacketLossForRole(host *models.Host, clusterRoleReqs *models.ClusterHostRequirements, hosts []*models.Host, inventoryCache InventoryCache) (ValidationStatus, []HostTimingMetric, error) {
```

Comment:
why do you need to create two functions that call the same function with a different parameter? 
Is there a reason not to expose the parameter?